src/java.base/unix/native/libjava/TimeZone_md.c

```diff
@@ -592,7 +592,21 @@ getGMTOffsetID()
 {
     time_t offset;
     char sign, buf[32];
-    offset = timezone;
+    struct tm localtm;
+    time_t clock;
+
+    clock = time(NULL);
+    if (localtime_r(&clock, &localtm) == NULL) {
+        return strdup("GMT");
+    }
+
+    struct tm gmt;
+
+    if (gmtime_r(&clock, &gmt) == NULL) {
+        return strdup("GMT");
+    }
+
+    offset = (gmt.tm_hour - localtm.tm_hour)*3600 + (gmt.tm_min - localtm.tm_min)*60;
```

Comment:
Since it is not using `timezone` anymore, we can reverse the subtraction, i.e., `localtime` - `gmtime` so that the weird sign switch below can be eliminated.

Reply:
I've reversed it and reverted the weird sign switching below this code snippet.